x-pack/plugins/data_quality/server/routes/get_index_mappings.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { schema } from '@kbn/config-schema';
+import { IRouter } from '@kbn/core/server';
+import { transformError, buildResponse } from '@kbn/securitysolution-es-utils';
+
+import { fetchMappings } from '../lib';
+import { GET_INDEX_MAPPINGS } from '../../common/constants';
+
+export const getIndexMappingsRoute = (router: IRouter) => {
+  router.get(
+    {
+      path: GET_INDEX_MAPPINGS,
+      validate: {
+        params: schema.object({
+          index_name: schema.string(),
```

Comment:
Our current plan (in the Data Quality dashboard) is to only request a specific index name, e.g. `.ds-logs-endpoint.alerts-default-2022.11.17-000001` from this route, but given that the underlying ES `mappings` API also accepts a pattern, e.g. `.ds-logs-*`, consider renaming the `request.params.index_name` parameter to `request.params.pattern`, to align the parameter name with the full capabilities of the API.